src/manifolds/SymplecticStiefel.jl

```diff
@@ -0,0 +1,554 @@
+    p_symplectic = rand(Symplectic(2n), hamiltonian_norm)
+    return canonical_projection(M, p_symplectic)
+end
+
+@doc raw"""
+    rand(::SymplecticStiefel{n, k}, p)
+
+The symplectic Stiefel tangent space at ``p`` can be parametrized as [^Bendokat2021]
+````math
+    T_p\operatorname{SpSt}(2n, 2k) = \{X = pΩ + p^sB \;|\;
+        Ω ∈ ℝ^{2k × 2k}, Ω^+ = -Ω,
+        p^s ∈ \operatorname{SpSt}(2n, 2(n- k)), B ∈ ℝ^{2(n-k) × 2k}, \},
+````
+where ``Ω \in `` is Hamiltonian and ``p^s`` means the symplectic complement of ``p`` s.t.
+``p^{+}p^{s} = 0``.
+
+To then generate random tangent vectors at ``p``, we set ``B = 0`` and generate a random
+Hamiltonian matrix ``Ω``.
+"""
+function Base.rand(::SymplecticStiefel{n,k}, p::AbstractMatrix) where {n,k}
+    Ω = rand_hamiltonian(Symplectic(2k))
+    return p * Ω
+end
+
+@doc raw"""
+    canonical_projection(::SymplecticStiefel, p_Sp)
+    canonical_projection!(::SymplecticStiefel{n,k}, p, p_Sp)
+
+Define the canonical projection from ``\operatorname{Sp}(2n, 2n)`` onto
+``\operatorname{SpSt}(2n, 2k)``, by projecting onto the first ``k`` columns
+and the ``n + 1``'th onto the ``n + k``'th columns [^Bendokat2021].
+
+It is assumed that the point ``p`` is on ``\operatorname{Sp}(2n, 2n)``.
+"""
+function canonical_projection(M::SymplecticStiefel{n,k}, p_Sp) where {n,k}
+    p_SpSt = similar(p_Sp, (2n, 2k))
+    return canonical_projection!(M, p_SpSt, p_Sp)
+end
+
+function canonical_projection!(::SymplecticStiefel{n,k}, p, p_Sp) where {n,k}
+    p[:, (1:k)] .= p_Sp[:, (1:k)]
+    p[:, ((k + 1):(2k))] .= p_Sp[:, ((n + 1):(n + k))]
+    return p
+end
+
+# compute p^+q (which is 2kx2k) in place of A
+function symplectic_inverse_times(M::SymplecticStiefel{n,k}, p, q) where {n,k}
+    A = similar(p, (2k, 2k))
+    return symplectic_inverse_times!(M, A, p, q)
+end
+
+function symplectic_inverse_times!(::SymplecticStiefel{n,k}, A, p, q) where {n,k}
+    # we write p = [p1 p2; p3 p4] (and q, too), then
+    p1 = @view(p[1:n, 1:k])
+    p2 = @view(p[1:n, (k + 1):(2k)])
+    p3 = @view(p[(n + 1):(2n), 1:k])
+    p4 = @view(p[(n + 1):(2n), (k + 1):(2k)])
+    q1 = @view(q[1:n, 1:k])
+    q2 = @view(q[1:n, (k + 1):(2k)])
+    q3 = @view(q[(n + 1):(2n), 1:k])
+    q4 = @view(q[(n + 1):(2n), (k + 1):(2k)])
+    A1 = @view(A[1:k, 1:k])
+    A2 = @view(A[1:k, (k + 1):(2k)])
+    A3 = @view(A[(k + 1):(2k), 1:k])
+    A4 = @view(A[(k + 1):(2k), (k + 1):(2k)])
+    mul!(A1, p4', q1)           # A1  = p4'q1
+    mul!(A1, p2', q3, -1, 1)    # A1 -= p2'p3
+    mul!(A2, p4', q2)           # A2  = p4'q2
+    mul!(A2, p2', q4, -1, 1)    # A2 -= p2'q4
+    mul!(A3, p1', q3)           # A3  = p1'q3
+    mul!(A3, p3', q1, -1, 1)    # A3 -= p3'q1
+    mul!(A4, p1', q4)           # A4  = p1'q4
+    mul!(A4, p3', q2, -1, 1)    #A4  -= p3'q2
+    return A
+end
+
+@doc raw"""
+    Compute the exponential mapping from eq. 3.19 of Bendokat-Zimmermann.
+"""
+function exp!(M::SymplecticStiefel{n,k}, q, p, X) where {n,k}
```

Comment:
Please add the complete formula here.

Reply:
The complete formula has been added to documentation now.

Reply:
Ah, and this documented the mutating exp!` as well, it should document `exp` instead.

Reply:
Ah now it is ok :)